docs/compute-resources.md

```diff
@@ -52,6 +52,263 @@ requirements for `Step`containers, they must be treated as if they are running i
 Tekton adjusts `Step` resource requirements to comply with [LimitRanges](#limitrange-support).
 [ResourceQuotas](#resourcequota-support) are not currently supported.
 
+Instead of specifying resource requirements on each `Step`, users can choose to specify resource requirements at the Task-level. If users specify a Task-level resource request, it will ensure that the kubelet reserves only that amount of resources to execute the `Task`'s `Steps`.
+If users specify a Task-level resource limit, no `Step` may use more than that amount of resources.
+
+Each of these details is explained in more depth below.
+
+Some points to note:
+
+- Task-level resource requests and limits do not apply to sidecars which can be configured separately.
+- Users may not configure the Task-level and Step-level resource requirements (requests/limits)  simultaneously.
+
+### Configuring Task-level Resource Requirements
```

Comment:
Since this PR just introduces the new field to "TaskRun", let's stick with only documenting that for now. Also, the examples should use "computeResources" rather than "resources".

Reply:
yap. (didn't update the doc yet. let me add the field back for PipelineRun and update the doc with our current TEP requirements)